### ISSUE TITLE
MQTT messages are not published

### ISSUE THREAD

**Issue by oroce** [fr3ts0n/AndrOBD-Plugin#2]
I configured the server, port, username, and password for my cloudmqtt instance (i used a non secure connection to the mqtt server)  and no message was published even though the app was connected to the car. 
I read in #1 that I push `Perform Action` then it should publish message as well but it did not happen.

Then I installed to laptop a mqtt server, I used [mosca](https://www.npmjs.com/package/mosca) (`npm i -g mosca pino && mosca --very-verbose | pino`). I changed the server, username, password parameters, I pushed the `Perform Action` button and still no message.

After that I tried to double verify that the mqtt server is working properly, so I installed a [mqtt client](https://www.npmjs.com/package/mqtt#cli) and with that I could see the published message: `mqtt pub -t AndrOBD -h '192.168.1.120' -m 'from mqtt.js'`.

Since the mqtt server is working, I can publish messages from different I can see those published if I'm subscribed to the given topics, it must be something the AndrOBD. I changed the logging level of the app to `ALL` but there was no about failured publish. I can attach if you want but I didn't see anything related to the mqtt plugin.

**Comment by keredoc** [comment]
I am also unable to see anything published. As stated above, all credentials, IP addresses of the broker, etc are correct.

**Comment by fr3ts0n** [comment]
Hi @oroce and @keredoc ,

Thank you for your report, and sorry for my late answer.

There is a setting in the plugin to select the PID data items which are to be published to the MQTT broker. (Plugin settings -> Data items)
This setting basically contains a list of all data items which have been received by the plugin from the AndrOBD app. 

See Screenshot: 
https://github.com/fr3ts0n/AndrOBD-Plugin/blob/master/fastlane/metadata/android/en-US/images/phoneScreenshots/Screenshot_4.png 

Initially this list is empty after installation of the plugin, since there was no data item sent to the plugin yet.
By default the setting of newly received data items from AndrOBD is NOT to publish them. 
(This is done by intension to ensure that only desired values are forwarded to the broker)

Please could you try following sequence:
1. Start AndrOBD
2. Select OBD data mode (for several seconds to have all data items sent to plugin at least once)
3. Select "Plugin manager" -> MQTT publisher -> CONFIGURE
4. In plugin settings:
  4.1 Select "Data items"
  4.2 Select data items to be published
  4.3 Wait for configured "Update period" time in seconds

Your selected data items should now get published to the broker every x seconds where x is the configured time of Plugin setting "Update period".

I again re-tested this on the latest versions and confirm this to work in my environment running a mosquitto MQTT broker.

Please let me know if this solves your problem.

Thank you

**Comment by oroce** [comment]
thanks @fr3ts0n for the detailed answer.

I'm gonna double check this, but checking the data items was the first thing I did.
![screenshot_20180618-223843](https://user-images.githubusercontent.com/462848/41560809-8597e016-7348-11e8-8c20-5c173d9d5c53.jpg)

**Comment by oroce** [comment]
Okay it seems I was able to track down what's going on in the background.

Even though the AndrOBD log was empty, the abd log (`adb logcat`) has the following entries:

```
09-16 12:17:58.294  4841  4917 I Thread  : Publish data
09-16 12:17:58.585  4841  4917 E MQTT    : Publish
09-16 12:17:58.585  4841  4917 E MQTT    : Not authorized to connect (5)
09-16 12:17:58.585  4841  4917 E MQTT    : 	at org.eclipse.paho.client.mqttv3.internal.ExceptionHelper.createMqttException(ExceptionHelper.java:28)
09-16 12:17:58.585  4841  4917 E MQTT    : 	at org.eclipse.paho.client.mqttv3.internal.ClientState.notifyReceivedAck(ClientState.java:885)
09-16 12:17:58.585  4841  4917 E MQTT    : 	at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:118)
09-16 12:17:58.585  4841  4917 E MQTT    : 	at java.lang.Thread.run(Thread.java:764)
```

I've been using CloudMQTT (they have free plan, so anyone can try it out: https://customer.cloudmqtt.com/instance/create?plan=cat), where I'm using:

- username
- password
- hostname
- custom port

I doublechecked the credentials, port, hostname everything is right, so probaby AndrOBD's MQTT plugin is the one which isn't passing down the proper parameters to Paho.

@fr3ts0n is there anything I could do to help tracking down/resolving this bug?

**Comment by oroce** [comment]
@fr3ts0n I was able to fix this issue, see #5
